### PR TITLE
MAINT: Simplify lookfor function

### PR DESCRIPTION
**Pull request by alexhenrie**
This is exactly the kind of thing for/else is for.

### CODE REVIEW

File: numpy/lib/utils.py [hunk 1]
```diff
@@ -788,13 +788,11 @@ def lookfor(what, module=None, import_modules=True, regenerate=False,
...
             continue
-        ok = True
         doc = docstring.lower()
         for w in whats:
             if w not in doc:
-                ok = False
                 break
-        if ok:
+        else:
             found.append(name)
```
Comment (eric-wieser) [inline numpy/lib/utils.py]: If we're going to touch this rarely used code, I'd be tempted to change this to:
```
if all(w in doc for w in whats):
    found.append(name)
```
which is even clearer than the `for` / `else` construction

Comment (alexhenrie) [reply numpy/lib/utils.py]: Great, let's do that instead. I just force-pushed a new commit.

**Review by eric-wieser — APPROVED** [review]

**Comment by mattip** [comment]
Not sure what is happening with coverage, but the UI does not seem useful in this case.

**Comment by mattip** [comment]
Thanks @alexhenrie

**Comment by alexhenrie** [comment]
Thank you!

**Pull request merged**